test/common/index.js

```diff
@@ -60,6 +60,31 @@ const hasOpenSSL3 = hasCrypto &&
 
 const hasQuic = hasCrypto && !!process.config.variables.openssl_quic;
 
+function parseTestFlags(filename = process.argv[1]) {
+  // The copyright notice is relatively big and the flags could come afterwards.
+  const bytesToRead = 1500;
+  const buffer = Buffer.allocUnsafe(bytesToRead);
+  const fd = fs.openSync(filename, 'r');
+  const bytesRead = fs.readSync(fd, buffer, 0, bytesToRead);
+  fs.closeSync(fd);
+  const source = buffer.toString('utf8', 0, bytesRead);
+
+  const flagStart = source.indexOf('// Flags: --') + 10;
+
+  if (flagStart === 9) {
+    return [];
+  }
+  let flagEnd = source.indexOf('\n', flagStart);
+  // Normalize different EOL.
+  if (source[flagEnd - 1] === '\r') {
+    flagEnd--;
+  }
+  return source
+    .substring(flagStart, flagEnd)
+    .replace(/_/g, '-')
+    .split(' ');
+}
```

Comment:
Is this just cleanup refactoring, moving this into its own function for clarity? If there’s a meaningful change to this code or the function it came from, what is it?

Reply:
I have extracted this logic so it can be used inside `snap.js`